### PR TITLE
Added flexibility to pull desired channel for Quay operator deployment

### PR DESCRIPTION
**Pull request by PrasadDesala**
Fixes: #7725

### CODE REVIEW

**Comment by PrasadDesala** [comment]
Quay test is currently failing due to an open issue in the Quay product. Therefore, I am providing the local run logs where the test is using the latest Quay version 3.8 for reference.

```
--------------------------------------------------------------- live log call ---------------------------------------------------------------
15:29:45 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc --kubeconfig /home/prasad/cluster/auth/kubeconfig -n openshift-marketplace get PackageManifest quay-operator -n openshift-marketplace -o yaml
15:29:46 - MainThread - ocs_ci.ocs.quay_operator - INFO  - Installing Quay operator: quay-operator
15:29:46 - MainThread - ocs_ci.ocs.resources.ocs - INFO  - Adding Subscription with name quay-operator
15:29:46 - MainThread - ocs_ci.utility.templating - INFO  - apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: quay-operator
  namespace: openshift-operators
spec:
  channel: stable-3.8
  installPlanApproval: Automatic
  name: quay-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace

15:29:46 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-operators create -f /tmp/Subscriptionogfax8fu -o yaml

```

**Review by Shrivaibavi — APPROVED** [review]
LGTM

File: ocs_ci/ocs/quay_operator.py [hunk 1]
```diff
@@ -47,13 +47,41 @@ def __init__(self):
             else constants.DEFAULT_STORAGECLASS_RBD
         )
 
-    def setup_quay_operator(self):
+    def get_quay_default_channel(self):
```
Comment (vavuthu) [inline ocs_ci/ocs/quay_operator.py:50]: you can get default channel from here: https://github.com/red-hat-storage/ocs-ci/blob/f87828ce9fd737524f5b49b42fd63affccc6f012/ocs_ci/ocs/resources/packagemanifest.py#L101

Comment (PrasadDesala) [reply ocs_ci/ocs/quay_operator.py:51]: Done

**Review by vavuthu — APPROVED** [review]

**Comment by PrasadDesala** [comment]
```
auth_api create_auth {'role': 'admin', 'system': 'noobaa', 'email': 'admin@noobaa.io', 'password': 'O2jdIH6iljs+oZ0hcfae3Q=='}
--------------------------------------------------------------- live log call ---------------------------------------------------------------
11:35:02 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc --kubeconfig /home/prasad/cluster/auth/kubeconfig -n openshift-marketplace get packagemanifest quay-operator -n openshift-marketplace -o yaml
11:35:03 - MainThread - ocs_ci.ocs.quay_operator - INFO  - Installing Quay operator: quay-operator
11:35:03 - MainThread - ocs_ci.ocs.resources.ocs - INFO  - Adding Subscription with name quay-operator
11:35:03 - MainThread - ocs_ci.utility.templating - INFO  - apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: quay-operator
  namespace: openshift-operators
spec:
  channel: stable-3.8
  installPlanApproval: Automatic
  name: quay-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace

```

**Review by vavuthu — APPROVED** [review]

**Pull request merged**